hasjob/views/helper.py

```diff
@@ -320,9 +320,68 @@ def bgroup(jobpost_ab, post):
     return (jobpost_ab.get(post.id) or cointoss()) if post.headlineb else None
 
 
-def cache_viewcounts(posts):
+def get_jobpost_impressions(jobpost_id):
+    return db.session.query(db.func.count(db.func.distinct(EventSession.user_id)).label('count')).join(
+        JobImpression).filter(JobImpression.jobpost_id == jobpost_id).first().count
+
+
+def get_post_viewcounts(jobpost_id):
+    cache_key = JobPost.viewcounts_key(jobpost_id)
+    values = g.viewcounts.get(cache_key)
+    jobpost = None
+    if not values:
+        values = redis_store.hgetall(cache_key)
```

Comment:
It is indeed always a dict: https://github.com/andymccurdy/redis-py/blob/master/redis/client.py#L406

Please indicate so in a comment.